ApplicationLibCode/Application/Tools/RiaLogging.cpp

```diff
@@ -163,6 +163,35 @@ void RiaDefaultConsoleLogger::writeToConsole( const std::string& str )
 #endif
 }
 
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RiaLogging::setLastMessage( const QString& message )
+{
+#pragma omp critical( critical_section_logging )
+    sm_lastMessage     = message;
+    sm_lastMessageTime = std::chrono::high_resolution_clock::now();
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RiaLogging::isSameMessage( const QString& message )
+{
+    if ( message == sm_lastMessage )
+    {
+        auto now      = std::chrono::high_resolution_clock::now();
+        auto duration = std::chrono::duration_cast<std::chrono::milliseconds>( now - sm_lastMessageTime );
+
+        if ( duration.count() < 1000 )
+        {
+            return true;
+        }
+    }
```

Comment:
This block also needs to be guarded with `critical pragma`, I think.

Reply:
Fixed